src/Metadata/Resource/ResourceMetadataCollection.php

```diff
@@ -86,6 +88,11 @@ public function getOperation(?string $operationName = null, bool $forceCollectio
         $this->handleNotFound($operationName, $metadata);
     }
 
+    public function isDynamic(): bool
+    {
+        return str_starts_with($this->resourceClass, self::DYNAMIC_RESOURCE_CLASS_PREFIX);
```

Comment:
imo `resourceClass` should be a `class-string` this breaks it.

Reply:
Sure but we need a way to distinguish a dynamic resource for a normal one.